profiles/serializer/normalizer.py

```diff
@@ -12,6 +12,11 @@ def normalize(value: Any) -> Any:
     return value
 
 
+@singledispatch
+def normalize_restricted(value: Any) -> Any:
```

Comment:
is the additional layer of indirection a type checker/linter requirement? (allows `Any` rather than just `Profile`?)

Reply:
I tried to follow the current pattern in place. `@singledispatch` is a way of using Generics (overloading) in python. Meaning, a call to `normalize` will first check the type of the first argument and, if another registered function exists that expects that type, then that implementation will be used.